src/call/mod.rs

```diff
@@ -764,27 +715,23 @@ impl SinkBase {
 
 #[cfg(test)]
 mod tests {
-    use grpc_sys;
-    use std::mem;
-
     use super::*;
 
-    unsafe fn make_message_reader(source: &[u8], nslice: usize) -> MessageReader {
-        let mut slices = Vec::with_capacity(nslice);
-        for _ in 0..nslice {
-            let len = source.len();
-            let mut slice = grpc_sys::grpc_slice_from_copied_buffer(source.as_ptr() as _, len);
-            slices.push(slice);
+    fn make_message_reader(source: &[u8], n_slice: usize) -> MessageReader {
+        let mut slices = Vec::with_capacity(n_slice);
+        for _ in 0..n_slice {
+            slices.push(From::from(source));
```

Comment:
why can't this simply be 

```rust
let mut slices = vec![source.into(); n_slice];
```

Reply:
I don't know there's such API, will refactor later.